aptos-move/framework/aptos-framework/sources/aptos_account.move

```diff
@@ -22,13 +42,37 @@ module aptos_framework::aptos_account {
         coin::register<AptosCoin>(&signer);
     }
 
+    /// Convenient function to transfer APT to a recipient account that might not exist.
+    /// This would create the recipient account first, which also registers it to receive APT, before transferring.
     public entry fun transfer(source: &signer, to: address, amount: u64) {
```

Comment:
nit: should we call this function by a more explicit name? smth like `transfer_to_existing_or_nonexisting_account` - this does sound way more verbose than `transfer`, but think `transfer` is a simple name that people might mis-use

Reply:
That's really verbose. I'm not sure how people would misuse transfer here as it's meant to implicitly create an account (and may be register the coin type) if one doesn't exist yet. As proposed (I'll create an AIP soon), this new flow/function should be the default to use so users wouldn't need to worry about the nuances of accounts and coin registration.

Reply:
yeah I don't feel strong about this, just thought that some devs might use the transfer function here as the default transfer function without realizing that they could be transferring coins to an account that doesn't exist yet. as long as we make it clear, sgtm